### ISSUE TITLE
Failure to serialize 1D polynomial WCS with domain

### ISSUE THREAD

**Issue by jehturner** [spacetelescope/gwcs#306]
@chris-simpson is working with some 1D co-ordinates and ran into a serialization failure that's illustrated by the following simplified example:
```python
from astropy.modeling.models import Chebyshev1D
from asdf import AsdfFile
from gwcs import WCS as gWCS, CoordinateFrame, SpectralFrame

f1 = CoordinateFrame(naxes=1, axes_order=(0,), axes_type=('SPATIAL',))
f2 = SpectralFrame()
m1 = Chebyshev1D(2, domain=(0, 1))

wcs = gWCS([(f1, m1),
            (f2, None)])

af = AsdfFile({'WCS': wcs})
```
```
Traceback (most recent call last):
  File "test.py", line 19, in <module>
    af = AsdfFile({'WCS': wcs})
  File "/home/jturner/repos/asdf/asdf/asdf.py", line 155, in __init__
    self.tree = tree
  File "/home/jturner/repos/asdf/asdf/asdf.py", line 404, in tree
    self._validate(asdf_object, custom=bool(tree))
  File "/home/jturner/repos/asdf/asdf/asdf.py", line 434, in _validate
    schema.validate(tagged_tree, self, reading=reading)
  File "/home/jturner/repos/asdf/asdf/schema.py", line 622, in validate
    validator.validate(instance, _schema=(schema or None))
  File "/home/jturner/anaconda3_2019.10/envs/wavecal/lib/python3.6/site-packages/jsonschema/validators.py", line 348, in validate
    raise error
jsonschema.exceptions.ValidationError: {'polynomial_type': 'chebyshev', 'coefficients': {'shape': [3], 'source': 0, 'datatype': 'float64', 'byteorder': 'little'}, 'domain': (0, 1), 'window': (-1, 1)} is not valid under any of the given schemas

Failed validating 'anyOf' in schema['properties']['steps']['items']['properties']['transform']:
    OrderedDict([('description',
                  'The transform from this step to the next one.  The\n'
                  'last step in a WCS should not have a transform, but\n'
                  'exists only to describe the frames and units of the\n'
                  'final output axes.\n'),
                 ('anyOf',
                  [OrderedDict([('$ref',
                                 'tag:stsci.edu:asdf/transform/transform-1.1.0')]),
                   OrderedDict([('type', 'null')])]),
                 ('default', None)])

On instance['steps'][0]['transform']:
    {'coefficients': {'byteorder': 'little',
                      'datatype': 'float64',
                      'shape': [3],
                      'source': 0},
     'domain': (0, 1),
     'polynomial_type': 'chebyshev',
     'window': (-1, 1)}
```
I don't understand the behaviour 100% (eg. altering either the model or the co-ordinate frames can make it go away), but I think this is happening because `gwcs/step-1.0.0.yaml` references the old `asdf/transform/transform-1.1.0`, which still references the deprecated `domain-1.0.0` that interferes with the recently-added serialization of polynomial domains (cf. https://github.com/astropy/astropy/pull/9906#issuecomment-580788361)?

We were apparently able to resolve this by changing `transform-1.1.0` to `transform-1.2.0` in `gwcs/step-1.0.0.yaml`; is that the correct solution, @nden? Thanks.

**Comment by jehturner** [comment]
PS. I think we didn't see this previously because we were using compound (and/or 2D) models, which go through a different schema such as `concatenate`, which is at `1.2.0` and references `transform-1.2.0`, while the top-level compound model itself doesn't have a domain.

**Comment by nden** [comment]
@jehturner You are right. The transform reference in gwcs needs to be updated. And the required asdf version needs to be bumped to 2.6.

**Comment by nden** [comment]
Resolved by spacetelescope/gwcs#307
